### ISSUE TITLE
Wrong assumptions on SCTP data chunk and design issue..

### ISSUE THREAD

**Issue by zecke** [google/gopacket#347]
In one SCTP package we can have _multiple_ SCTPData chunks. The current code doesn't support it and will provide all other chunks as payload. I removed the special case of decodeSCTPChunk and then put a Payload into the SCTPData struct. One issue is that the test data is kind of invalid...

The fundamental issue is that the structure of SCTP is like this.

```
   IP
  SCTP
  SCTPDataChunk
     Payload
  SCTPDataChunk
     Payload
```

While p.NextDecoder will just be linear. We would need to have some kind of special "Payload" so we can jump to the next SCTPDataChunk. Alternatively be always eager for SCTP...

**Comment by gconnell** [comment]
Would it make sense for SCTPDataChunk to just have a []byte to its data, then it could return the next SCTPDataChunk immediately after?

**Comment by zecke** [comment]
Yes, that would be the easiest (and how I patched gopacket) locally. But I don't know if "Payload" has significance for other users gopacket? E.g. if code should work transparently for tcp and sctp then someone might have written something like the below code? Should this continue to work?

```
for _, layer := ranges packet.Layers() {
    if payload, ok := packet.(Payload); !ok {
      continue
    }
}
```
